recipes/botan/all/conanfile.py

```diff
@@ -0,0 +1,308 @@
+import os
+from conans import ConanFile, tools
+from conans.errors import ConanException, ConanInvalidConfiguration
+from conans.model.version import Version
+
+
+class BotanConan(ConanFile):
+    name = 'botan'
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "https://github.com/randombit/botan"
+    license = "BSD-2-Clause"
+    exports = ["patches/*"]
+    description = "Botan is a cryptography library written in C++11."
+    settings = 'os', 'arch', 'compiler', 'build_type'
+    options = {
+        'amalgamation': [True, False],
+        'bzip2': [True, False],
+        'debug_info': [True, False],
+        'openssl': [True, False],
+        'quiet': [True, False],
+        'shared': [True, False],
+        'fPIC': [True, False],
+        'single_amalgamation': [True, False],
+        'sqlite3': [True, False],
+        'zlib': [True, False],
+        'boost': [True, False],
+        'enable_modules': "ANY",
+        'system_cert_bundle': "ANY"
+    }
+    default_options = {'amalgamation': True,
+                       'bzip2': False,
+                       'debug_info': False,
+                       'openssl': False,
+                       'quiet': True,
+                       'shared': True,
+                       'fPIC': True,
+                       'single_amalgamation': False,
+                       'sqlite3': False,
+                       'zlib': False,
+                       'boost': False,
+                       'enable_modules': None,
+                       'system_cert_bundle': None}
+
+    def configure(self):
+        compiler = self.settings.compiler
+        version = Version(self.settings.compiler.version.value)
+
+        if self.settings.os == "Windows" and compiler == "Visual Studio" and version < "14":
+            raise ConanInvalidConfiguration("Botan doesn't support MSVC < 14")
+
+        elif compiler == "gcc" and version >= "5" and compiler.libcxx != "libstdc++11":
```

Comment:
This is not accurate, as 5.0 with `libstdc++` is fine. Starting from 5.1 it's not.
@uilianries is there way to get the minor position of the compiler version? Since settings.compiler.version will only tell me `5`, even for gcc 5.1, I cannot distinguish between 5.0 and 5.1.

Reply:
this is weird because GCC5 is backward compatible. About the minor, actually you should use the version which is present in the settings. There some functions in Conan which are able to detect the minor version, but you can't use here because the compatibility (as internal method, we can't guarantee its behavior/structure in the future), you can take a look by searching for detect_defaults_settings in Conan code base.

Reply:
Oof, nevermind. [There is no GCC 5.0](https://gcc.gnu.org/releases.html). So checking for `version >= 5` is exactly what we want to do here :sweat_smile: